### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by jwtong** [JuniorTour/vue-template-babel-compiler#3]
invalid expression: Unexpected token '.' in

    optional?.chaining

v1.0.0

**Comment by JuniorTour** [comment]
@jwtong Thank yor for your feedback

Could you provide more context? So that we can help you ASAP.

For example:
1. Your usage, `Vue-cli, Nuxt.js, Webpack` etc.
2. Your steps or code snippets whiclh got the error above.
3. A minimal reproduction project is prefer, eg: [vue-template-babel-compiler-nuxt-project](https://github.com/JuniorTour/vue-template-babel-compiler-nuxt-project)

**Comment by jwtong** [comment]
@JuniorTour This was from the Nuxt demo project actually (vue-template-babel-compiler-nuxt-project), and is consistent with my other project I am trying to set this up for.  `"nuxt": "^2.15.7", "vue-template-babel-compiler": "^1.0.0"`

**Comment by Intrflex** [comment]
Im currently having the same error with the vuecli installation currently unable to get it to work

**Comment by JuniorTour** [comment]
@jwtong Sorry for late reply.

Is your problem solved?

I just test [Nuxt.js DEMO project](https://github.com/JuniorTour/vue-template-babel-compiler-nuxt-project) again.

It will throw the same error when not config `nuxt.config.js`:
![ZI$4NLF%)_DDOT0SST13FVV](https://user-images.githubusercontent.com/14243906/131238661-cfdc6ee5-2b94-4523-8d9c-d6509e9f2525.png)

## So the solution could be add `build.loaders.vue` to your  `nuxt.config.js`:
``` js
// nuxt.config.js
export default {
  // Build Configuration: https://go.nuxtjs.dev/config-build
  build: {
    loaders: {
      vue: {
        compiler: require('vue-template-babel-compiler')
      }
    },
  },
  // ...
}
```

If the problem still not be solved, feel free to ask my again.

**Comment by jwtong** [comment]
@JuniorTour Yes, so it now works with that config, just from upgrading to 1.0.2. There is now just a SSR issue, I know someone else opened a PR for that as well.

https://github.com/JuniorTour/vue-template-babel-compiler/issues/7

**Comment by JuniorTour** [comment]
@jwtong Thank you for your help! 

It is a bug indeed, I just published `1.0.3` to fix it.  

### You can try it by:
``` sh
npm install vue-template-babel-compiler@1.0.3 -D
// or yarn
yarn add vue-template-babel-compiler@1.0.3 -D
```

Now this compiler will work for `SSR` too!

![8b40e54c646984a92edb58ac2fcf7b4](https://user-images.githubusercontent.com/14243906/132031108-c09e0267-a5c0-4a9a-9eb1-f152dc11de7c.png)

I also update [vue-template-babel-compiler-nuxt-project](https://github.com/JuniorTour/vue-template-babel-compiler-nuxt-project/commit/cd2ee97fd7a9404381707c951a1afe66bdac3bf9)

Thank you for your feedback again!

**Comment by f3ltron** [comment]
Doesnt work with latest nuxt project

**Comment by JuniorTour** [comment]
> Doesnt work with latest nuxt project

@f3ltron Thanks for your feedback.

I just test by this [demo-nuxt-project](https://github.com/JuniorTour/vue-template-babel-compiler-nuxt-project) with `"nuxt": "^2.15.8"` and ` "vue-template-babel-compiler": "^1.0.6-0"`.

It works fine.

How do you use this lib? 

Could you provide more details? For example:
- Your `nuxt.config.js`
- Error message you got

I'd be glad to help you use this lib.

**Comment by zavaremre** [comment]
Hi, Having error with [demo-nuxt-project](https://github.com/JuniorTour/vue-template-babel-compiler-nuxt-project). I cloned the demo repo and started the project. then got the following error: 

![image](https://user-images.githubusercontent.com/35058780/151980084-0c37ad29-edb7-4a41-872b-940839a26fa7.png)

**Comment by JuniorTour** [comment]
> Hi, Having error with [demo-nuxt-project](https://github.com/JuniorTour/vue-template-babel-compiler-nuxt-project). I cloned the demo repo and started the project. then got the following error: 
> 
> It is working while building but it doesn't work with `npm run dev`
> Using node v12.22.4 
> Nuxt v2.15.7
> 

Currently, The `nullish coealasing (??)`  only work for `node.js v14+`.

So a temp workaround is:
- upgrade node.js to great than 14.
- delete ?? usage

I'm still look into can we use it below v14, welcome for working toghther.

Thanks for your feedback.

Related:
- https://github.com/JuniorTour/vue-template-babel-compiler/issues/29

**Comment by Songoo7** [comment]
npm Installed to vue 2 webpack project, added "compiler: require('vue-template-babel-compiler'),"

But code still crash on .?   
Anybody who got this working ? Are there some additional installation settings needed ?

**Comment by JuniorTour** [comment]
> npm Installed to vue 2 webpack project, added "compiler: require('vue-template-babel-compiler'),"
> 
> But code still crash on .?   
> Anybody who got this working ? Are there some additional installation settings needed ?

@Songoo7 

What is your Nodejs version? Currently this lib need nodejs v14 aboce.

Could your provide a reproduction repo or online demo?

It would be my pleasure to help you.

**Comment by Songoo7** [comment]
> > npm Installed to vue 2 webpack project, added "compiler: require('vue-template-babel-compiler'),"
> > But code still crash on .?
> > Anybody who got this working ? Are there some additional installation settings needed ?
> 
> @Songoo7
> 
> What is your Nodejs version? Currently this lib need nodejs v14 aboce.
> 
> Could your provide a reproduction repo or online demo?
> 
> It would be my pleasure to help you.

Hello , I have node v16.15, webpack 3, babel 6, vue 2.4.4 .

**Comment by JuniorTour** [comment]
@Songoo7 What is the version of `vue-loader` you are using?

If version is smaller than `v15.0.0`, the [compiler option](https://vue-loader.vuejs.org/options.html#compiler) is not valid. It only valid for `vue-loader` >= `v15.0.0`.

[vue-loader v14](https://vue-loader-v14.vuejs.org/en/options.html#compilermodules) does NOT have this option to use `vue-template-babel-compiler`.

**Comment by Songoo7** [comment]
"vue-loader": "^13.0.5", so I should raise to 15 ? I'll try if it let me.

**Comment by Songoo7** [comment]
I got to vue-loader 14.0.0. But sadly going to 15 is impossible for me :( tried to follow https://vue-loader.vuejs.org/migrating.html#notable-breaking-changes and https://vue-loader.vuejs.org/guide/#vue-cli . But those guides are not enough detailed for this. Added new loader, added new less sass configs installed less-loader, sass-loader but in the end got an hundreds and hundreds red errors :) .
